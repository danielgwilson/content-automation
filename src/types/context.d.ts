export interface IContext {
  outputDir: string;
  resourceDir: string;
  debug: boolean;
}
