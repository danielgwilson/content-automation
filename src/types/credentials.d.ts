export interface ICredentials {
  username: string;
  password: string;
}
