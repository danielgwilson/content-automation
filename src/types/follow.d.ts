export interface IFollowOutput {
  sessionStart: Date;
  sessionEnd: Date;
  context: import("./context").IContext;
  account: string;
  tags: string[];
  actionsTaken: IFollowAction[];
  manager: {
    proxy;
    executablePath;
    timeout;
  };
}

export interface IFollowAction {
  timestamp: Date;
  target: IAccount;
}

export interface IAccount {
  username: string;
  following: number;
  followers: number;
  likes: number;
}
