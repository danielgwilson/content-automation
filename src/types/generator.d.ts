export interface IGeneratorOutput {
  id: string;
  dateGenerated: Date;
  elapsedTime: number;

  media: { metadata: any; thumbnail: any; render: IRenderOutput };
}

export interface IRenderOutput {
  job: any;
  settings: {
    workpath: string;
    maxMemoryPercent: number;
    skipCleanup: boolean;
    debug: boolean;
  };
}
