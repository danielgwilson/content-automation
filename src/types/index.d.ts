export { IContext } from "./context";
export { IPost, IProcessedPost } from "./post";
