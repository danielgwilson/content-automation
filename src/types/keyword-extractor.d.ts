type Language =
  | "danish"
  | "dutch"
  | "english"
  | "french"
  | "galician"
  | "german"
  | "italian"
  | "polish"
  | "portuguese"
  | "romanian"
  | "russian"
  | "spanish"
  | "swedish";

declare module "keyword-extractor" {
  function extract(
    str: string,
    options?: {
      remove_digits: boolean;
      return_changed_case: boolean;
      return_chained_words: boolean;
      language: Language;
      remove_duplicates: boolean;
      return_max_ngrams: boolean;
    }
  ): string[];
  function getStopwords(options: { language?: Language }): string[];
}
