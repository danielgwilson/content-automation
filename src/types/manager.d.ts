interface IManagerOutput {
  proxy?: import("./proxy").IProxy;
  executablePath?: string;
  timeout?: number;
  browserType: string;
}
