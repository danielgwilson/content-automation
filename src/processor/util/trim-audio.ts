import { IPostSection } from "../../types";
import { getAudioLengthForSections } from "../sections";

export function trimAudio(
  sections: IPostSection[],
  options: { maxAudioLength?: number; speakingRate?: number } = {}
) {
  const { maxAudioLength, speakingRate } = options;
  if (!maxAudioLength) return sections;

  const trimmedSections: IPostSection[] = [];
  let totalAudioLength = 0;
  for (let section of sections) {
    const sectionAudioLength = getAudioLengthForSections([section]);
    if (totalAudioLength + sectionAudioLength >= maxAudioLength) {
      break;
    }
    trimmedSections.push(section);
    totalAudioLength += sectionAudioLength;
  }

  if (trimmedSections.length <= 1) {
    const minimumTotalAudioLength = getAudioLengthForSections(
      sections.slice(0, 2)
    );
    let errorText = `Failed to trim audio output for processed sections within maxAudioLength of ${maxAudioLength}s; minimum audio length: ${minimumTotalAudioLength}`;
    if (speakingRate) {
      const requiredSpeakingRate = Math.round(
        ((minimumTotalAudioLength - 2) / (maxAudioLength - 2)) *
          (speakingRate * 100)
      );
      errorText = errorText.concat(
        `\nRequired speakingRate: ${requiredSpeakingRate}`
      );
    }
    throw new Error(errorText);
  }

  return trimmedSections;
}
