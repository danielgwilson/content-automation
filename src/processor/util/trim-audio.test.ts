import { trimAudio } from "./trim-audio";
import { IPostSection } from "../../types";

describe("Trim Audio", () => {
  it("Trims audio to max length parameter", () => {
    const maxAudioLength = 60; // 60 seconds worth of audio
    const sections = [
      {
        type: "title",
        fragments: [
          {
            text: "Why do you like to be alive?",
            textWithPriors: "Why do you like to be alive?",
            audio: {
              filePath:
                "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-0.0.mp3",
              fileName: "1fadf0f4-3dad-4cdf-95c2-3a424e218214-0.0.mp3",
              length: 10,
              voice: {
                languageCode: "en-US",
                name: "en-US-Wavenet-B",
                ssmlGender: "MALE",
              },
              audioConfig: { audioEncoding: "MP3", speakingRate: 1 },
            },
          },
        ],
        length: 10,
        score: 124447,
        author: "skopein",
        gildings: { silver: 214, gold: 19, platinum: 13 },
        children: [],
      },
      {
        type: "comment",
        fragments: [
          {
            text:
              "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too.",
            textWithPriors:
              "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too.",
            audio: {
              filePath:
                "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.0.mp3",
              fileName: "1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.0.mp3",
              length: 60 * 16,
              voice: {
                languageCode: "en-US",
                name: "en-US-Wavenet-B",
                ssmlGender: "MALE",
              },
              audioConfig: { audioEncoding: "MP3", speakingRate: 1 },
            },
          },
          {
            text:
              "However, PLEASE keep in mind rule 7 and stop begging for awards, because a lot of you are doing it and it is against the rules.",
            textWithPriors:
              "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too. However, PLEASE keep in mind rule 7 and stop begging for awards, because a lot of you are doing it and it is against the rules.",
            audio: {
              filePath:
                "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.1.mp3",
              fileName: "1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.1.mp3",
              length: 8.208,
              voice: {
                languageCode: "en-US",
                name: "en-US-Wavenet-B",
                ssmlGender: "MALE",
              },
              audioConfig: { audioEncoding: "MP3", speakingRate: 1 },
            },
          },
          {
            text:
              "\nAdditionally, if anyone is reading this thread and struggling, there are some resources available:",
            textWithPriors:
              "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too. However, PLEASE keep in mind rule 7 and stop begging for awards, because a lot of you are doing it and it is against the rules. \nAdditionally, if anyone is reading this thread and struggling, there are some resources available:",
            audio: {
              filePath:
                "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.2.mp3",
              fileName: "1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.2.mp3",
              length: 5.928,
              voice: {
                languageCode: "en-US",
                name: "en-US-Wavenet-B",
                ssmlGender: "MALE",
              },
              audioConfig: { audioEncoding: "MP3", speakingRate: 1 },
            },
          },
        ],
        length: 0,
        score: 1,
        author: "-eDgAR-",
        gildings: { silver: 6 },
        children: [],
      },
    ] as IPostSection[];

    const trimmedSections = trimAudio(sections, { maxAudioLength });

    expect(trimmedSections.length).toBe(2);
    expect(trimmedSections).toMatchSnapshot();
  });
});
