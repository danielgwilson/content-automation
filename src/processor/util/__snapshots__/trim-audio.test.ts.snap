// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trim Audio Trims audio to max length parameter 1`] = `
Array [
  Object {
    "author": "skopein",
    "children": Array [],
    "fragments": Array [
      Object {
        "audio": Object {
          "audioConfig": Object {
            "audioEncoding": "MP3",
            "speakingRate": 1,
          },
          "fileName": "1fadf0f4-3dad-4cdf-95c2-3a424e218214-0.0.mp3",
          "filePath": "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-0.0.mp3",
          "length": 10,
          "voice": Object {
            "languageCode": "en-US",
            "name": "en-US-Wavenet-B",
            "ssmlGender": "MALE",
          },
        },
        "text": "Why do you like to be alive?",
        "textWithPriors": "Why do you like to be alive?",
      },
    ],
    "gildings": Object {
      "gold": 19,
      "platinum": 13,
      "silver": 214,
    },
    "length": 10,
    "score": 124447,
    "type": "title",
  },
  Object {
    "author": "-eDgAR-",
    "children": Array [],
    "fragments": Array [
      Object {
        "audio": Object {
          "audioConfig": Object {
            "audioEncoding": "MP3",
            "speakingRate": 1,
          },
          "fileName": "1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.0.mp3",
          "filePath": "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.0.mp3",
          "length": 960,
          "voice": Object {
            "languageCode": "en-US",
            "name": "en-US-Wavenet-B",
            "ssmlGender": "MALE",
          },
        },
        "text": "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too.",
        "textWithPriors": "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too.",
      },
      Object {
        "audio": Object {
          "audioConfig": Object {
            "audioEncoding": "MP3",
            "speakingRate": 1,
          },
          "fileName": "1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.1.mp3",
          "filePath": "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.1.mp3",
          "length": 8.208,
          "voice": Object {
            "languageCode": "en-US",
            "name": "en-US-Wavenet-B",
            "ssmlGender": "MALE",
          },
        },
        "text": "However, PLEASE keep in mind rule 7 and stop begging for awards, because a lot of you are doing it and it is against the rules.",
        "textWithPriors": "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too. However, PLEASE keep in mind rule 7 and stop begging for awards, because a lot of you are doing it and it is against the rules.",
      },
      Object {
        "audio": Object {
          "audioConfig": Object {
            "audioEncoding": "MP3",
            "speakingRate": 1,
          },
          "fileName": "1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.2.mp3",
          "filePath": "/Users/danielgwilson/local_git/reddit-youtube-video-bot/temp/1fadf0f4-3dad-4cdf-95c2-3a424e218214/voice-over/1fadf0f4-3dad-4cdf-95c2-3a424e218214-1.2.mp3",
          "length": 5.928,
          "voice": Object {
            "languageCode": "en-US",
            "name": "en-US-Wavenet-B",
            "ssmlGender": "MALE",
          },
        },
        "text": "
Additionally, if anyone is reading this thread and struggling, there are some resources available:",
        "textWithPriors": "Yes everyone, this post has a lot of a awards and many people in this thread have gotten awards too. However, PLEASE keep in mind rule 7 and stop begging for awards, because a lot of you are doing it and it is against the rules. 
Additionally, if anyone is reading this thread and struggling, there are some resources available:",
      },
    ],
    "gildings": Object {
      "silver": 6,
    },
    "length": 0,
    "score": 1,
    "type": "comment",
  },
]
`;
